micro-rdk-server/esp32/main.rs

```diff
@@ -39,44 +36,46 @@ mod esp32 {
     }
 
     pub(crate) fn main_esp32() {
-        micro_rdk::esp32::esp_idf_svc::sys::link_patches();
-
-        micro_rdk::esp32::esp_idf_svc::log::EspLogger::initialize_default();
+        esp_idf_svc::sys::link_patches();
+        esp_idf_svc::log::EspLogger::initialize_default();
 
-        let mut r = Box::<ComponentRegistry>::default();
-        register_examples(&mut r);
-        let repr = RobotRepresentation::WithRegistry(r);
+        #[cfg(feature = "qemu")]
+        let _network = {
+            use micro_rdk::esp32::esp_idf_svc::hal::prelude::Peripherals;
+            log::info!("creating eth object");
+            let sys_loop = esp_idf_svc::eventloop::EspEventLoop::take().unwrap();
+            let eth = EspEth::wrap(
+                EthDriver::new_openeth(Peripherals::take().unwrap().mac, sys_loop.clone()).unwrap(),
+            )
+            .unwrap();
+            eth_configure(&sys_loop, eth).unwrap()
+        };
 
         {
-            micro_rdk::esp32::esp_idf_svc::sys::esp!(unsafe {
-                micro_rdk::esp32::esp_idf_svc::sys::esp_vfs_eventfd_register(
-                    &micro_rdk::esp32::esp_idf_svc::sys::esp_vfs_eventfd_config_t { max_fds: 5 },
+            esp_idf_svc::sys::esp!(unsafe {
```

Comment:
not particularly. It was more about moving this network stuff up rather than the eventfd stuff down.
Will move it higher. 

Any reason we have these braces around the `eventfd_register` call? it's already being conversted to a result and unwrapped, so not sure why it's scoped.